src/components/EnvelopeQuickActions.vue

```diff
@@ -0,0 +1,80 @@
+<!--
+  - SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+  - SPDX-License-Identifier: AGPL-3.0-or-later
+-->
+
+<template>
+	<div class="list-item-content__quick-actions">
+		<EmailUnread v-if="isRead"
+			:size="22"
+			:title="t('mail', 'Mark as unread')"
+			@click.prevent="$emit('toggle-seen')" />
+		<EmailRead v-else
+			:size="22"
+			:title="t('mail', 'Mark as read')"
+			@click.prevent="$emit('toggle-seen')" />
+		<ImportantIcon v-if="isImportant"
+			:size="22"
+			:title="t('mail', 'Mark as unimportant')"
+			@click.prevent="$emit('toggle-important')" />
+		<ImportantOutlineIcon v-else
+			:size="22"
+			:title="t('mail', 'Mark as important')"
+			@click.prevent="$emit('toggle-important')" />
+		<IconDelete :size="22"
+			:title="t('mail', 'Delete thread')"
+			@click.prevent="$emit('delete')" />
+	</div>
+</template>
+
+<script>
+
+import ImportantOutlineIcon from 'vue-material-design-icons/LabelVariantOutline.vue'
+import EmailUnread from 'vue-material-design-icons/EmailOutline.vue'
+import EmailRead from 'vue-material-design-icons/EmailOpenOutline.vue'
+import IconDelete from 'vue-material-design-icons/DeleteOutline.vue'
+import ImportantIcon from 'vue-material-design-icons/LabelVariant.vue'
+
+export default {
+	name: 'EnvelopeQuickActions',
+	components: {
+		EmailRead,
+		EmailUnread,
+		ImportantIcon,
+		ImportantOutlineIcon,
+		IconDelete,
+	},
+	props: {
+		isRead: {
+			type: Boolean,
+			default: false,
+		},
+		isImportant: {
+			type: Boolean,
+			default: false,
+		},
+	},
+	data() {
+
+	},
+
+}
+</script>
+
+<style lang="scss" scoped>
+.list-item-content__quick-actions {
+	display: none;
```

Comment:
I'm not an a11y expert but this feels like it could be tricky for screen readers

Reply:
I looked into this, screen readers [don't read things with `display: none`](https://www.nomensa.com/blog/how-improve-web-accessibility-hiding-elements/), meaning that users with a screen reader won't be able to see the single click actions, seeing as I don't think they have the ability to hover.

Having said that, I wouldn't consider it a problem since all the actions are still accessible from the actions menu.